bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/IDFUtil.java

```diff
@@ -97,6 +108,16 @@
 				+ IDFConstants.IDF_TOOLS_SCRIPT;
 		return new File(idf_py_script);
 	}
+	
+	/**
+	 * @return idf_tools.py file path based on the IDF_PATH given in the argument
+	 */
+	public static File getIDFToolsScriptFile(String idf_path)
+	{
+		String idf_py_script = idf_path + IPath.SEPARATOR + IDFConstants.TOOLS_FOLDER + IPath.SEPARATOR
+				+ IDFConstants.IDF_TOOLS_SCRIPT;
+		return new File(idf_py_script);
+	}
```

Comment:
Same as above, you could call this arg method from the existing implementation to avoid duplicate code.

```
public static File getIDFPythonScriptFile()
	{
		String idf_path = getIDFPath();
		return getIDFPythonScriptFile(idf_path);
	}

```